build.gradle

```diff
@@ -935,6 +935,9 @@ def calculateVersion() {
 
   if (project.hasProperty('version') && (project.version =~ calVerPattern)) {
     return "${project.version}"
+  } else if (project.hasProperty('versionprefix') && (project.versionprefix =~ calVerPattern)) {
+    def gitDetails = getGitCommitDetails(7) // Adjust length as needed
+    return "${project.versionprefix}-${gitDetails.hash}"
```

Comment:
instead of introducing versionprefix, could we instead add a 'force githash suffix' property?  The behavior would be more intuitive imo, instead of having `version` and `versionprefix`

Reply:
Yeah I think that would be a good way of achieving the same thing. I'm on vacation tomorrow but I'll update the PR on Monday.

Reply:
I've refactored along the lines you suggested @garyschult. I've also included commented-out examples of using them.

The example `version` field I've put in the file includes an example vendor name: `24.5.6-acme`. That's how we'll probably use the field, and I think other vendors might want to set it in a similar way if they package Besu into their own offerings, so their users can check that they're running a version from their vendor. Obviously just setting `24.5.6` works fine as well.